Assets/Scripts/Audio/AudioWaveformDrawer.cs

```diff
@@ -0,0 +1,317 @@
+                int startSample = Mathf.FloorToInt(x * (_dataSamples.Length / (float)_waveformTexture.width));
+                int endSample = Mathf.Min(startSample + (_dataSamples.Length / _waveformTexture.width), _dataSamples.Length);
+                float sum = 0;
+
+                for (int j = startSample; j < endSample; j++)
+                    sum += Mathf.Abs(_dataSamples[j]); //get the absolute value of the sample
+
+                float averageSample = sum / (_dataSamples.Length / _waveformTexture.width);
+                float scaledAverage = averageSample * _heightScaleModifier; //scale the average sample value
+
+                for (int y = 0; y < _waveformTexture.height; y++) //set the pixel color based on the average sample value
+                    _textureColors[x + y * _waveformTexture.width] = (y < scaledAverage) ? _renderColor : Color.clear;
+            }
+
+            _waveformTexture.SetPixels(_textureColors);
+            _waveformTexture.filterMode = FilterMode.Point; //set the filter mode to point for a pixelated look
+            _waveformTexture.Apply();
+
+            GameObject drawerObject = Instantiate(_drawerPrefab, transform.position, _drawerPrefab.transform.rotation);
+            drawerObject.transform.SetParent(transform);
+
+            drawerObject.transform.localScale = new Vector3(_textureWidth / _renderDownScaleModifier, 1, _textureHeight / _renderDownScaleModifier);
+            drawerObject.GetComponent<Renderer>().material.mainTexture = _waveformTexture;
+        }
+
+        private void Update()
+        {
+            _waveformPositionOffset.y = _curserYPosition; //update yOffset for possible live tweaking
+
+            CheckAudioControls(); //update controls for manipulating the audio track
+            UpdateCursorPosition(); //move cursor along audio waveform texture
+        }
+
+        /// <summary>
+        /// Handles all input checks related to manipulating the audio track.
+        /// </summary>
+        private void CheckAudioControls()
+        {
+            if (!Input.GetKey(KeyCode.LeftControl)) //can't hold left ctrl, messes with other keybinds
+            {
+                if (Input.GetKey(_forwardKey)) //scrub forward in song
+                    audioSource.time = Mathf.Clamp(audioSource.time + timeScrubAmount, 0, _audioClipDuration - 1);
+
+                if (Input.GetKey(_backwardKey)) //scrub backward in song
+                    audioSource.time = Mathf.Clamp(audioSource.time - timeScrubAmount, 0, _audioClipDuration);
+            }
+
+            if (Input.GetKeyDown(_pauseKey)) //pausing song
+            {
+                if (!isPaused)
+                {
+                    audioSource.Pause();
+                }
+                else
+                {
+                    audioSource.UnPause();
+                    if (!audioSource.isPlaying)
+                        audioSource.Play();
+                }
```

Comment:
```suggestion
                if (!isPaused && audioSource.IsPlaying())
                {
                    audioSource.Pause();
                }
                else if(!audioSource.IsPlaying())
                {
                    audioSource.UnPause();
                    if (!audioSource.isPlaying)
                        audioSource.Play();
                }
```

Reply:
Again, I don't think this is necessary, I only need to know if the user has paused the track.